java/dagger/hilt/android/plugin/src/main/kotlin/dagger/hilt/android/plugin/HiltGradlePlugin.kt

```diff
@@ -34,6 +36,28 @@ class HiltGradlePlugin : Plugin<Project> {
     val androidExtension = project.extensions.findByType(BaseExtension::class.java)
       ?: throw error("The Hilt Android Gradle plugin can only be applied to an Android project.")
```

Comment:
To avoid enforcing plugin application order, you can switch to `project.plugins.withType(com.android.build.gradle.api.AndroidBasePlugin::class.java) { configureHilt() }`

Reply:
can we detect that it is applied to a project w/o android since it is likely to be a mistake? should we just add a hook to after evaluate to ensure we've configured something ?

Reply:
Yep. Pull a `var configured = false` out and then set to true in the block. In `afterEvaluate` throw if it's still false.

Reply:
thanks. did that.